torch/quantization/quantize.py

```diff
@@ -123,25 +125,27 @@ def add_quant_dequant(module):
         module._modules[name] = add_quant_dequant(child)
     return module
 
-def prepare(module, qconfig_dict=None):
-    r"""Prepares the module for calibration or training given a qconfig_dict.
-    Note that the module will be modified inplace but in case the input module
-    is a leaf module, a wrapped module will be returned.
+def prepare(model, qconfig_dict=None):
+    r"""Prepares the model for calibration or training given a qconfig_dict.
+    Note that the model will be modified inplace but in case the input model
+    is a leaf model, a wrapped model will be returned.
 
     Args:
-        mod: input module
+        mod: input model
         qconfig_dict: dictionary that maps from name of submodule to quantization
                       configuration
     Return:
-        A module with qconfig propogated, observer and quant dequant or fake
-        quant modules attached, a module that is ready for calibration or
+        A model with qconfig propogated, observer and quant dequant or fake
+        quant modules attached, a model that is ready for calibration or
         training
     """
-    propagate_qconfig(module, qconfig_dict)
+    propagate_qconfig(model, qconfig_dict)
     if qconfig_dict:
-        module = add_quant_dequant(module)
-    add_observer(module)
-    return module
+        model = add_quant_dequant(model)
+    add_observer(model)
+    if model.training:
```

Comment:
it feels to fragile to rely on it. Top-level different functions for qat and pqt feel better. In the end, you'd still need to pass very different "eval" function into them

Reply:
I thought we want a one line API for both, no?
I guess the reason we have eval_fn in the argument of top level API is to allow unification and this is applicable for graph mode as well

Reply:
For prepare, the changes required to support QAT and PTQ are small. Why not try to have the same API for QAT and PTQ?.

Reply:
I don't feel strongly about it. But that's why it feels weird - the qconfig one passes is going to be different for QAT and PTQ. So it's more logical for prepare() to just enforce (or switch) model to be in the right eval/train state. In this case we have two switches which need to work together.

Reply:
qconfig pass is the same, we just pass in different qconfig_dict.

Reply:
do you mean change API to `prepare(model, training=False, qconfig_dict=None)`?

Reply:
I don't know how this extra boolean is different than just asking user to always specify whether the model is in train mode or inference mode before calling quantize

Reply:
will split the APIs.